test/realm_set_test.dart

```diff
@@ -277,7 +286,10 @@ Future<void> main([List<String>? args]) async {
       realm.write(() {
         realm.add(testSet);
         var set = testSet.setByType(type).set;
-        set.add(values.first);
+        expect(set.add(values.first), true);
+
+        //adding an already existing value is a no operation
+        expect(set.add(values.first), false);
```

Comment:
Looking at this I'm wondering if it would be valuable to test add with upsert option for the parent object. Imagine we want to create a set with managed objects that are already added in the realm and then we create the parent with the set.

Reply:
I will add a such test